test/framework/flux.go

```diff
@@ -247,6 +291,123 @@ func (e *ClusterE2ETest) initGit(ctx context.Context) {
 	e.GitClient = g.Client
 }
 
+func (e *ClusterE2ETest) workloadClusterConfGitPath(w *WorkloadCluster) string {
+	return e.clusterConfGitPathFromName(w.ClusterName)
+}
+
+func (e *ClusterE2ETest) workloadClusterConfigGitPath(w *WorkloadCluster) string {
+	return filepath.Join(e.GitWriter.Dir(), e.workloadClusterConfGitPath(w))
+}
+
+func (e *ClusterE2ETest) buildWorkloadClusterConfigFileForGit(w *WorkloadCluster) {
+	b := w.generateClusterConfigYaml()
+	g := e.GitWriter
+	p := filepath.Dir(e.workloadClusterConfigGitPath(w))
+
+	if _, err := os.Stat(p); errors.Is(err, os.ErrNotExist) {
+		err := os.MkdirAll(p, os.ModePerm)
+		if err != nil {
+			w.T.Fatalf("Creating directory [%s]: %v", g.Dir(), err)
+		}
+	}
+
+	_, err := g.Write(e.workloadClusterConfGitPath(w), b, filewriter.PersistentFile)
+	if err != nil {
+		w.T.Fatalf("Error writing cluster config file to local git folder: %v", err)
+	}
+}
+
+func (e *ClusterE2ETest) addWorkloadClusterConfigToGit(ctx context.Context, w *WorkloadCluster) error {
+	p := e.workloadClusterConfGitPath(w)
+	g := e.GitClient
+	if err := g.Add(p); err != nil {
+		return fmt.Errorf("adding cluster path changes at %s: %v", p, err)
+	}
+
+	if err := e.pushStagedChanges(ctx, "EKS-A E2E Flux test workload configuration changes added"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (e *ClusterE2ETest) deleteWorkloadClusterConfigFromGit(ctx context.Context, w *WorkloadCluster) error {
+	p := e.workloadClusterConfGitPath(w)
+	g := e.GitClient
+	if err := g.Remove(p); err != nil {
+		return fmt.Errorf("removing cluster config at path %s: %v", p, err)
+	}
+
+	if err := e.pushStagedChanges(ctx, "EKS-A E2E Flux test workload configuration deleted"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (e *ClusterE2ETest) pushStagedChanges(ctx context.Context, commitMessage string) error {
+	g := e.GitClient
+	if err := g.Commit(commitMessage); err != nil {
```

Comment:
Can we maybe add these commit & push operations to a separate helper method to use both here & in the existing `pushConfigChanges` method? Seems like it's doing the same thing at that point

Reply:
What do you mean?

The way I see it, `pushedStagedChanges` is the helper method since it includes the common commit and push flow in both. I could update `pushConfigChanges` to use it

Reply:
The current lines 348-359 in this file are basically the same as this: https://github.com/aws/eks-anywhere/blob/main/test/framework/flux.go#L861-L872

So I wanted your thoughts on maybe extracting those steps into their own little helper method to avoid duplicating code